lib/temporal/workflow/future.rb

```diff
@@ -71,7 +71,7 @@ def cancel
 
       private
 
-      attr_reader :context, :cancelation_id, :result, :exception
+      attr_reader :context, :cancelation_id, :result
```

Comment:
So we already have `get` exposed, which will return `exception` in case of a failure. Do you think we also need to expose `exception`? This would probably suggest that we should expose `result` following the same logic

Reply:
I could go either way here. If I remove `exception`, then I'll have to modify `.execute!` to no longer use it, but this would be a small change. Generally, I'm a fan of having methods that only return one kind of object. `.get` is handy for awaiting a result, but the behavior in `.execute` is different from `.execute!` because of the way exceptions are handled.

I'll take another pass at getting this more consistent.

Reply:
I've returned `:exception` to private. Overall, this looks simpler with the more narrow interface.